docs/sources/awssns.md

```diff
@@ -1,126 +1,107 @@
-> ```
->
-> Alternatively, credentials can be used as literal strings instead of references to Kubernetes Secrets by replacing
-> `valueFrom` attributes with `value` inside API objects' manifests.
+A fully qualified ARN is required to uniquely identify the AWS SNS topic.
 
-### As a AWSSNSSource object
+![SNS topic](../images/awssns-source/sns-topic.png)
 
-Copy the sample manifest from `config/samples/awssnssource.yaml` and replace the pre-filled `spec` attributes with the
-values corresponding to your _AWS SNS_ topic. Then, create that `AWSSNSSource` object in your Kubernetes cluster:
+As shown in the above screenshot, you can obtain the ARN of a SNS topic from the AWS console. It typically has the following format:
 
-```console
-$ kubectl -n <my_namespace> create -f my-awssnssource.yaml
+```
+arn:aws:sns:{awsRegion}:{awsAccountId}:{topicName}
 ```
 
-### As a ContainerSource object
-
-Copy the sample manifest from `config/samples/awssns-containersource.yaml` and replace the pre-filled environment
-variables under `env` with the values corresponding to your _AWS SNS_ topic. Then, create that `ContainerSource` object
-in your Kubernetes cluster:
+Alternatively you can also use the [AWS CLI][aws-cli]. The following command retrieves the ARN of a SNS topic named `MyQueue` in the `us-west-2` region.
 
 ```console
-$ kubectl -n <my_namespace> create -f my-awssns-containersource.yaml
+$ aws --region us-west-2 sns list-topics
+{
+    "Topics": [
+        ...
+        {
+            "TopicArn": "arn:aws:sns:us-west-2:123456789012:MyTopic"
+        },
+        ...
+    ]
+}
 ```
 
-### As a Deployment object bound by a SinkBinding
+### API Credentials
+
+The TriggerMesh AWS SNS event source authenticates calls to the AWS SNS API using an [Access Key][accesskey]. The page at this link contains instructions to create an access key when signed either as the root user or as an IAM user. Take note of the **Access Key ID** and **Secret Access Key**, they will be used to create an instance of the event source.
+
+It is considered a [good practice][iam-bestpractices] to create dedicated users with restricted privileges in order to programmatically access AWS services. Permissions can be added or revoked granularly for a given IAM user by attaching [IAM Policies][iam-policies] to it.
+
+As an example, the following policy contains the permissions required by the TriggerMesh AWS SNS event source to read and delete messages from any topic linked to the AWS account:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": [
+                "sns:Subscribe",
+                "sns:ConfirmSubscription"
+            ],
+            "Resource": "arn:aws:sns:*:123456789012:e2e-*"
+        },
+        {
+            "Sid": "VisualEditor1",
+            "Effect": "Allow",
+            "Action": "sns:Unsubscribe",
+            "Resource": "*"
+        }
```

Comment:
I think we should either rename or remove those `Sid` attributes. "VisualEditor" doesn't provide much value.

Reply:
oh right.. fixing

Reply:
updating rather...

Reply:
please take another look